src/main/java/com/conveyal/taui/analysis/broker/EC2RequestConfiguration.java

```diff
@@ -88,40 +88,48 @@ public String toString() {
         }
     }
 
-    RequestSpotInstancesRequest prepareSpotRequest(){
-        LaunchSpecification launchSpecification = new LaunchSpecification()
-                .withImageId(AnalysisServerConfig.workerAmiId)
-                .withInstanceType(AnalysisServerConfig.workerInstanceType)
-                .withSubnetId(AnalysisServerConfig.workerSubnetId)
-                .withIamInstanceProfile(new IamInstanceProfileSpecification().withArn(AnalysisServerConfig.workerIamRole))
-                .withUserData(userDataScript)
-                .withEbsOptimized(true);
+    RequestSpotInstancesRequest prepareSpotRequest (int nSpot, String clientToken) {
+        RequestSpotLaunchSpecification launchSpecification = RequestSpotLaunchSpecification.builder()
+                .imageId(AnalysisServerConfig.workerAmiId)
+                .instanceType(AnalysisServerConfig.workerInstanceType)
+                .subnetId(AnalysisServerConfig.workerSubnetId)
+                .iamInstanceProfile(IamInstanceProfileSpecification.builder().arn(AnalysisServerConfig.workerIamRole).build())
```

Comment:
Is this desired whitespace/indentation?

Reply:
Do you mean the two levels of indent rather than one (as seen in the later RunInstancesRequest builder method chain)?  My IDE code formatter seems to want to do two levels of indent here, while below I had manually done one level because it seemed more uniform. But I do find the two levels more readable. Do you have a preference?

Reply:
Two levels is good; it caught my eye based on strange formatting in the GH diff view, but it's actually fine.